Add singularity.libraryDir to config page

christopher-hakkaart opened the pull request:
Fixes [#1879](https://github.com/nextflow-io/nextflow/issues/1879)

pditommaso commented on docs/reference/config.md:
```diff
@@ -1375,6 +1375,9 @@ The following settings are available:
+`singularity.libraryDir`
+: Directory where remote Singularity images are retrieved. When using a computing cluster it must be a shared folder accessible to all compute nodes.
+
 `singularity.noHttps`
```
it should be added an entry for `apptainer.libraryDir` as well. 

Also the description could be extended to make it coherent with the explanation [here](https://www.nextflow.io/docs/latest/container.html)

> When looking for a Singularity image file, Nextflow first checks the library directory, and if the image file is not found, the cache directory is used as usual. The library directory can be defined either using the NXF_SINGULARITY_LIBRARYDIR environment variable or the singularity.libraryDir configuration setting (the latter overrides the former).

christopher-hakkaart commented on docs/container.md:
```diff
@@ -123,7 +123,7 @@ This feature requires the `apptainer` tool to be installed where the workflow ex
 
 :::{versionadded} 21.09.0-edge
-When looking for a Apptainer image file, Nextflow first checks the *library* directory, and if the image file is not found, the *cache* directory is used s usual. The library directory can be defined either using the `NXF_APPTAINER_LIBRARYDIR` environment variable or the `apptainer.libraryDir` configuration setting (the latter overrides the former).
+Nextflow first checks the library directory when retrieving Apptainer images. The cache directory is used if the image file is not found. The library directory can also be defined using the `NXF_APPTAINER_LIBRARYDIR` environment variable. The `apptainer.libraryDir` configuration option overrides `NXF_APPTAINER_LIBRARYDIR` environment variable if both are set.
```
They are separate pages. This is a part of a top-level paragraph. Below is a note within a reference list.

christopher-hakkaart commented on docs/reference/config.md:
```diff
@@ -52,6 +52,13 @@ The following settings are available:
+: Directory where remote Apptainer images are retrieved. When using a computing cluster it must be a shared folder accessible to all compute nodes.
+
+  :::{versionadded} 21.09.0-edge
+  Nextflow first checks the library directory when retrieving Apptainer images. The cache directory is used if the image file is not found. The library directory can also be defined using the `NXF_APPTAINER_LIBRARYDIR` environment variable. The `apptainer.libraryDir` configuration option overrides `NXF_APPTAINER_LIBRARYDIR` environment variable if both are set.
```
This is a note within a reference list. It's specific to `apptainer.libraryDir` so it is indented to align.

pditommaso commented on docs/container.md:
```diff
@@ -123,7 +123,7 @@ This feature requires the `apptainer` tool to be installed where the workflow ex
 
 :::{versionadded} 21.09.0-edge
-When looking for a Apptainer image file, Nextflow first checks the *library* directory, and if the image file is not found, the *cache* directory is used as usual. The library directory can be defined either using the `NXF_APPTAINER_LIBRARYDIR` environment variable or the `apptainer.libraryDir` configuration setting (the latter overrides the former).
+Nextflow first checks the library directory when retrieving Apptainer images. The cache directory is used if the image file is not found. The library directory can also be defined using the `NXF_APPTAINER_LIBRARYDIR` environment variable. The `apptainer.libraryDir` configuration option overrides `NXF_APPTAINER_LIBRARYDIR` environment variable if both are set.
```
I feel this can be improved, it's not clear what's the difference with the cache directory. 

I would do something along the lines:

> Moreover Nextflow uses the library directory to determine the location of the Apptainer container file. Nextflow first checks the *library* directory, if the container file is not found, it check the cache directory. The main difference between the library directory and the cache directory is that the first is assumed to be a read-only container repository, while the latter is expected to be writable path where container files can added for caching purposes. 

> The library directory can be defined either using the `apptainer.libraryDir` configuration setting or the `NXF_APPTAINER_LIBRARYDIR` environment variable. The configuration file option overrides the environment variable if both are set.

christopher-hakkaart replied:
Thanks, I've worked this so that it explains each setting, and then their relationship. Let me know if it works.

pditommaso commented on docs/container.md:
```diff
@@ -123,7 +123,7 @@ This feature requires the `apptainer` tool to be installed where the workflow ex
 Nextflow caches those images in the `apptainer` directory in the pipeline work directory by default. However it is suggested to provide a centralised cache directory by using either the `NXF_APPTAINER_CACHEDIR` environment variable or the `apptainer.cacheDir` setting in the Nextflow config file.
 
 :::{versionadded} 21.09.0-edge
```
`versionadded` older than 2 years can be removed

christopher-hakkaart replied:
Manually removed.

christopher-hakkaart commented:
I reworked the text to describe each option, and then their relationship. I felt that this became a large piece of text that was too much for the reference material so I removed the note. Let me know if you want something else added in it's place.

pditommaso commented:
well done
